main.cpp

```diff
@@ -19,13 +19,22 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        auto cStyleCExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        if (cStyleCExpr) {
+            auto lPloc = cStyleCExpr->getLParenLoc();
+            auto rPloc = cStyleCExpr->getRParenLoc();
+            auto endLoc = cStyleCExpr->getEndLoc();
+
+            rewriter.ReplaceText(lPloc, 1, "static_cast<");
+            rewriter.ReplaceText(rPloc, 1, ">(");
+            rewriter.InsertTextAfterToken(endLoc, ")");

```

Comment:
Please describe the difference between 
`rewriter.InsertTextAfterToken(endLoc, ")")`  and `rewriter.InsertText(endLoc, ")")`.

Reply:
Method `insertTextAfterToken` always insert text after token. With `InsertText` you can insert text after or before buffer and you can specify indent lines.

Reply:
Examples required.

Reply:
`InsertTextAfterToken` insert text after token, which constains in param location, but `InsertText` insert text into location.
Example: token `func`, text for insert`my`. `InsertText` insert text before letter `f`, result `myfunc`, but `InsertTextAfterToken` after letter `c`, result `funcmy`.